### DISCUSSION TITLE
Check deployment status for specific commit

### DISCUSSION THREAD

**Discussion by danielbroll** [vercel/community#874] [Help]
Hey Vercel Team,

I would like to check the deployment status from within a github actions pipeline.

Most Vercel API endpoints depend on the _deploymentId_. How do I get the right deployment ID for my specific commit?

I could get all deployments for my team via https://api.vercel.com/v6/deployments?teamId=x
and then filter it manually via _deployment.meta.githubCommitSha_

Is there an easier way?

**Comment by amyegan** [comment]
Hi @danielbroll! If you have the deployment URL, then you can [get deployment details by URL](https://vercel.com/docs/rest-api#endpoints/deployments/get-a-deployment-by-id-or-url). The data returned from `/v13/deployments/{idOrUrl}` will include the deployment ID. The [create](https://vercel.com/docs/rest-api#endpoints/deployments/create-a-new-deployment) endpoint will also include the deployment id in its response. 

If you need to look it up by commit, then yes you would need to get a list of recent deployments and filter. You wouldn't need to pull all team deployments if you know more details that can help you filter. For example, you could get only the deployments for a single project in specified environment(s) that occurred within a date range. More options outlined here: [vercel.com/docs/rest-api#endpoints/deployments/list-deployments](https://vercel.com/docs/rest-api#endpoints/deployments/list-deployments)
